File: r/raindrops/raindrops.R
raindrops <- function(number) {
  is_divided_by_three <- number %% 3 == 0;
  is_divided_by_five <- number %% 5 == 0;
  is_divided_by_seven <- number %% 7 == 0;
  
  if(
    !is_divided_by_three &&
    !is_divided_by_five &&
    !is_divided_by_seven
  ) {
    return(toString(number))
  }

  sound <- ""

  if (is_divided_by_three) {
    sound = paste(sound, "Pling", sep = "")
  }

  if (is_divided_by_five) {
    sound = paste(sound, "Plang", sep = "")
  }

  if (is_divided_by_seven) {
    sound = paste(sound, "Plong", sep = "")
  }
  
  return(sound)
}

File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(num)
    is_divisible_by_three = (num % 3).zero?
    is_divisible_by_five = (num % 5).zero?
    is_divisible_by_seven = (num % 7).zero?

    if !is_divisible_by_three && !is_divisible_by_five && !is_divisible_by_seven
      return num.to_s
    end

    sound = ""

    if is_divisible_by_three
      sound.concat "Pling"
    end

    if is_divisible_by_five
      sound.concat "Plang"
    end

    if is_divisible_by_seven
      sound.concat "Plong"
    end

    return sound
  end
end
